staging/src/k8s.io/client-go/tools/remotecommand/remotecommand.go

```diff
@@ -140,3 +142,50 @@ func (e *streamExecutor) Stream(options StreamOptions) error {
 
 	return streamer.stream(conn)
 }
+
+// StreamContext opens a protocol streamer to the server and streams until a client closes
+// the connection or the server disconnects, but it can be closed through the Context.
+func (e *streamExecutor) StreamContext(ctx context.Context, options StreamOptions) error {
+	req, err := http.NewRequestWithContext(ctx, e.method, e.url.String(), nil)
+	//req, err := http.NewRequest(e.method, e.url.String(), nil)
+	if err != nil {
+		return fmt.Errorf("error creating request: %v", err)
+	}
+	conn, protocol, err := spdy.Negotiate(
+		e.upgrader,
+		&http.Client{Transport: e.transport},
```

Comment:
(I know you copied this but) http.Client caches connections, IIRC? are the connections flushed? It doesn't seem good to construct one of these every time, why is it done that way?

Reply:
Sorry, I’m still a novice, so I don’t know much.😢